src/App.tsx

```diff
@@ -1,26 +1,296 @@
+  const [visibleErr, setVisibleErr] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo[] | null>(null);
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+
+  const filteredTodos = (tod: Todo[], type: Status) => {
+    switch (type) {
+      case Status.active:
+        return tod.filter(todo => !todo.completed);
+      case Status.completed:
+        return tod.filter(todo => todo.completed);
+
+      default:
+        return tod;
+    }
+  };
+
+  const visibleTodos = tempTodo
+    ? filteredTodos(tempTodo, stat)
+    : filteredTodos(todos, stat);
+
+  const editSelectedInput = useRef<HTMLInputElement>(null);
+  const isAnyCompleted = todos.some(todo => todo.completed);
+
+  const resetErr = () =>
+    setTimeout(() => {
+      setVisibleErr(false);
+      setErrMessage('');
+    }, 3000);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      getTodos()
+        .then(setTodos)
+        .catch(() => {
+          setVisibleErr(true);
+          setErrMessage('Unable to load todos');
+          resetErr();
+        });
+    };
+
+    loadTodos();
+  }, []);
+
+  const updateTodo = async (updatedTodo: Todo, option: keyof Todo) => {
+    try {
+      setIsLoading(updatedTodo.id);
+      const todoToUpdate = todos.find(tod => tod.id === updatedTodo.id);
+
+      let newTodo: Todo = {
+        id: 0,
+        userId: 0,
+        title: '',
+        completed: false,
+      };
+
+      if (option === 'completed' && todoToUpdate) {
+        newTodo = { ...todoToUpdate, completed: !updatedTodo.completed };
+      }
+
+      if (option === 'title' && todoToUpdate) {
+        newTodo = { ...todoToUpdate, title: editedTitle };
+      }
+
+      await patchTodo(newTodo);
+
+      await getTodos().then(setTodos);
+
+      // setTodos(prevTodos => {
+      //   return prevTodos.map(todo => {
+      //     if (todo.id === updatedTodo.id) {
+      //       return { ...todo, ...newTodo };
+      //     }
+
+      //     return todo;
+      //   });
+      // });
+    } catch {
+      setVisibleErr(true);
+      setErrMessage('Unable to update a todo');
+      resetErr();
+    } finally {
+      setIsLoading(null);
+    }
+  };
+
+  const addTodo = async () => {
+    const trimedTitle = newTitle.trim();
+
+    if (trimedTitle === '') {
+      setVisibleErr(true);
+      setErrMessage('Title should not be empty');
+      resetErr();
+
+      return;
+    }
+
+    if (trimedTitle !== '') {
+      const newTodo: Todo = {
+        id: 0,
+        userId: 472,
+        title: trimedTitle,
+        completed: false,
+      };
+
+      const temp = {
+        id: 0,
+        userId: 472,
+        title: trimedTitle,
+        completed: false,
+      };
+
+      setTempTodo([...todos, temp]);
+
+      try {
+        setIsLoading(0);
+
+        await postTodo(newTodo).then(respond => {
+          setTempTodo(null);
+          setTodos(prevTodos => [...prevTodos, respond]);
+        });
+
+        await getTodos().then(setTodos);
```

Comment:
Bad practice! 
You never wants to refresh your list in that manner: having 2 API calls instead of 1 will quickly create performance issues. 

See, the logic here should be as simple as possible: 
user added a todo => you received a 200 response => you are 100% sure that the list is updated on the sever => no need to fetch it, you can change the UI already. 

Set a temp todo in order to make that loader work before success => replace it with the actual todo from response on success, that will pretty much do the job.